src/PrestaShopBundle/ApiPlatform/Pagination/PaginationElements.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\ApiPlatform\Pagination;
+
+class PaginationElements
+{
+    public function __construct(
+        public int $totalItems,
+        public ?string $orderBy,
+        public ?string $sortOrder,
+        public int $limit,
+        public ?int $offset,
+        public ?array $filters,
+        public array $items,
```

Comment:
```suggestion
        public readonly array $items,
        public readonly int $totalItems,
        public readonly int $limit,
        public readonly ?int $offset = null,
        public readonly ?array $filters = null,
        public readonly ?string $orderBy = null,
        public readonly ?string $sortOrder = null,
```
I wonder if we should make this DTO immutable by using the `readonly` attribute, also maybe the optional parameters should be nullable and placed last Even if e don't need in core implementation this may be handy for usage in modules